### PR TITLE
PV2-2067 query for successful submissions

### PR DESCRIPTION
**Pull request by carlbm**

### CODE REVIEW

File: src/SFA.DAS.Payments.Monitoring.Metrics.Function.AcceptanceTests/App.config [hunk 1]
```diff
@@ -1,17 +1,14 @@
...
-  <configSections>
-    <section name="specFlow" type="TechTalk.SpecFlow.Configuration.ConfigurationSectionHandler, TechTalk.SpecFlow" />
-  </configSections>
+  
   <connectionStrings>
     <add name="PaymentsConnectionString" connectionString="Data Source=.;Initial Catalog=SFA.DAS.Payments.Database;Integrated Security=True;Persist Security Info=False;Pooling=True;MultipleActiveResultSets=True;Connect Timeout=60;Encrypt=False;TrustServerCertificate=True" />
   </connectionStrings>
   <appSettings>
     <add key="ValidateSubmissionWindowFunctionUrl" value="http://localhost:7071/api/ValidateSubmissionWindow" />
+    <add key="SuccessfulSubmissionsFunctionUrl" value="http://localhost:7071/api/v1/Submission/Successful" />
```
Comment (tom-gough) [inline src/SFA.DAS.Payments.Monitoring.Metrics.Function.AcceptanceTests/App.config:9]: Is this a different function? Should it have it's own port? Is that even doable for local debug run?

**Review by tom-gough — DISMISSED** [review]

File: src/SFA.DAS.Payments.Monitoring.Metrics.Function.AcceptanceTests/App.config (hunk 1 above)
Comment (carlbm) [reply src/SFA.DAS.Payments.Monitoring.Metrics.Function.AcceptanceTests/App.config:9]: As I understand it there is one 'function app' with multiple endpoints. Not sure how to create multiple functions from one project, and one of the (later) requirements was that it shared the project with the existing function.

File: src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs [hunk 2]
```diff
@@ -19,6 +19,7 @@ public interface IMetricsQueryDataContext
 		DbSet<DataLockEventNonPayablePeriodModel> DataLockEventNonPayablePeriods { get; }
 		DbSet<RequiredPaymentEventModel> RequiredPaymentEvents { get; }
 		DbSet<PaymentModel> Payments { get; }
+		DbSet<LatestSuccessfulJobModel> LatestSuccessfulJobs { get; }
```
[the remark this answers is not in the document]
Comment (carlbm) [reply src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs:22]: Hmm. Not sure that that was my change? It is being used by other metrics queries!

File: src/SFA.DAS.Payments.Monitoring.Metrics.Application.UnitTests/Submission/SubmissionJobsServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            repository.Setup(x => x.GetLatestSuccessfulJobsForCollectionPeriod(academicYear, collectionPeriod))
+                .ReturnsAsync(testJobs);
+
+            var actual = await sut.SuccessfulSubmissionsForCollectionPeriod(academicYear, collectionPeriod);
+
+            foreach (var job in testJobs)
+            {
+                actual.SuccessfulSubmissionJobs.Should().ContainEquivalentOf(new {Ukprn = job.Ukprn});
+            }
+        }
```
[the remark this answers is not in the document]
Comment (carlbm) [reply src/SFA.DAS.Payments.Monitoring.Metrics.Application.UnitTests/Submission/SubmissionJobsServiceTests.cs:53]: I've left the one above as it was testing the mapping from DcJobId to JobId, but I've added an assertion for JobId to this test

**Review by hardik-desai-kcl — APPROVED** [review]

**Pull request merged**